ts/utils/navigation.ts

```diff
@@ -45,3 +50,22 @@ export function getCurrentRouteKey(navNode: any): string | undefined {
   }
   return undefined;
 }
+
+export function getRouteName(route: NavigationRoute): Option<string> {
+  if (route.index === undefined) {
+    return route.routeName ? some(route.routeName) : none;
+  }
+  if (route.index >= route.routes.length) {
+    return none;
+  }
+  return getRouteName(route.routes[route.index]);
+}
+
+export function whereAmIFrom(nav: NavigationHistoryState): Option<string> {
+  if (nav.length === 0) {
+    return none;
+  }
+  const navLength = nav.length;
+  const lastStep = nav[navLength - 1].routes[nav[navLength - 1].index];
```

Comment:
We should access an array in a safe way.
An improvement could be:

```typescript
import { index } from "fp-ts/lib/Array";

export function whereAmIFrom(nav: NavigationHistoryState): Option<string> {
  if (nav.length === 0) {
    return none;
  }
  const navLength = nav.length;
  return index(navLength - 1, [...nav]).fold(none, ln =>
    getRouteName(ln.routes[ln.index])
  );
}
```

Reply:
the edit is in 17ebbe4 I've removed the the if statement cause the case is already handled by the `fold` function